presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSourceFactory.java

```diff
@@ -340,7 +342,7 @@ public static ConnectorPageSource createOrcPageSource(
 
             List<HiveColumnHandle> physicalColumns = getPhysicalHiveColumnHandles(columns, useOrcColumnNames, reader.getTypes(), path);
 
-            if (!physicalColumns.isEmpty() && physicalColumns.stream().allMatch(hiveColumnHandle -> hiveColumnHandle.getColumnType() == AGGREGATED)) {
+            if (areFooterStatsReliable(fileSplit) && !physicalColumns.isEmpty() && physicalColumns.stream().allMatch(hiveColumnHandle -> hiveColumnHandle.getColumnType() == AGGREGATED)) {
```

Comment:
I believe this should not be from the customSplitInfo. This should be an another parameter like ,
            boolean appendRowNumberEnabled
            
 When we create the page from metalake or other places, we can pass in false and that would make it consistent.

Reply:
For metalake, there is a possibility that some splits may not have any mutation in which case we can still rely on the stats right?

Reply:
Chatted with Arun offline and its better to keep it consistent with already added feature for "appendRowNumber". 
Arun has a WIP PR to clean up HiveTableLayoutHandle and other files, I will wait for the PR to make this change. This will help me avoid touching many files to introduce this new boolean.